Makefile

```diff
@@ -0,0 +1,47 @@
+# Go parameters
+GOCMD=go
+GOBUILD=$(GOCMD) build
+GOCLEAN=$(GOCMD) clean
+GOTEST=$(GOCMD) test
+BINARY_NAME=nancy
+BUILD_VERSION_LOCATION=github.com/sonatype-nexus-community/nancy/buildversion
+GOLANGCI_VERSION=v1.24.0
+GOLANGCI_LINT_DOCKER=golangci/golangci-lint:$(GOLANGCI_VERSION)
+LINT_CMD=golangci-lint cache status --color always && golangci-lint run --timeout 5m --color always -v --max-same-issues 10
+
+all: deps test lint build
+
+.PHONY: lint clean deps env-setup build test integration-test package
+
+lint:
+	docker run --rm -v $$(pwd):/app -v $$(pwd)/.cache:/root/.cache -w /app $(GOLANGCI_LINT_DOCKER) /bin/sh -c "$(LINT_CMD)"
+
+ci-lint:
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin $(GOLANGCI_VERSION)
+	$(LINT_CMD)
+
+clean:
+	$(GOCLEAN)
+	rm -rf .cache
+	rm -f $(BINARY_NAME)
+	rm -f $(BINARY_NAME)-*amd64*
+
+deps:
+	$(GOCMD) mod download
+	$(GOCMD) mod verify
+	$(GOCMD) mod tidy
+
+env-setup:
+	export GO111MODULE=on CGO_ENABLED=0
```

Comment:
Does this work?
I didn't think exported vars persisted to other steps.


Testing with this Makefile leads me to believe it doesn't
```
stuff:
        export WHATEVER=blah

things: stuff
        env
        echo $(WHATEVER)
```

Reply:
Fixed here 3421af6

Reply:
I liked the idea.

Might make sense to make `GO_BUILD_FLAGS` a top level var if you want to keep the explicitness.... Or just :shipit:

Reply:
Done b4b18fb